workers/namespacegcworker.py

```diff
@@ -18,6 +19,13 @@ class NamespaceGCWorker(QueueWorker):
     """
 
     def process_queue_item(self, job_details):
+        try:
+            with GlobalLock("LARGE_GARBAGE_COLLECTION", lock_ttl=NAMESPACE_GC_TIMEOUT + 60):
```

Comment:
Does this need to be using the same lock as the repository gc worker?
This worker is performing a delete  on the DeletedNamespace table, does it need to be performed exclusively from the repo gc?

Reply:
Yes, since they could both GC the same repository, at the same time.